src/openforms/translations/middleware.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from django.conf import settings
+from django.http import HttpRequest
+from django.utils import translation
+
+from openforms.typing import RequestHandler
+from openforms.utils.urls import is_admin_request
+
+logger = logging.getLogger(__name__)
+
+
+def _remove_language_cookie_from_request(request: HttpRequest) -> None:
+    """
+    Produce a request instance without cookie-based language information.
+
+    settings.LANGUAGE_COOKIE_NAME
+    """
+    logger.debug(
+        "Removing the language cookie (if present) from request %r",
+        request,
+        extra={"request": request},
+    )
+    cookie_name = settings.LANGUAGE_COOKIE_NAME
+    if cookie_name in request.COOKIES:
+        del request.COOKIES[cookie_name]
+        logger.debug(
+            "Cookie %s was present but is now ignored.",
+            cookie_name,
+            extra={"request": request},
+        )
+
+
+class AdminLocaleMiddleware:
+    """
+    Discard the language cookie for locale-information.
+
+    While the language cookie is the primary mechanism to specify the desired language,
+    this applies to the public UI and not the admin. For the admin UI, we either use
+    the browser Accept-Language header or look up user preferences stored in the user
+    account itself.
+    """
+
+    def __init__(self, get_response: RequestHandler):
+        self.get_response = get_response
+
+    def __call__(self, request: HttpRequest):
+        # we do not language-code prefixed URLs
```

Comment:
I'm not entirely sure what is meant by "prefixed URLs"?

EDIT: judging from the changes to `is_admin_request`, I'm assuming prefixed with the admin base path

Reply:
django has a [`i18n_patterns`](https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#django.conf.urls.i18n.i18n_patterns) which essentially comes down to a `/nl/` or `/en/` prefix to have localized URLs. That's the sort of URLs this refers to :) We don't use them or aren't planning on doing so, so that's left out of scope for this feature.